pywr/parameters/_parameters.pyx

```diff
@@ -828,44 +828,64 @@ cdef class AggregatedParameter(AggregatedParameterBase):
         The aggregation function. Must be one of {"sum", "min", "max", "mean",
         "product"}, or a callable function which accepts a list of values.
     """
+    cpdef setup(self):
+        super(AggregatedParameter, self).setup()
+        assert(len(self.parameters))
 
-    cpdef double value(self, Timestep timestep, ScenarioIndex scenario_index) except? -1:
+    cpdef calc_values(self, Timestep timestep):
         cdef Parameter parameter
-        cdef double value, value2
-        assert(len(self.parameters))
+        cdef double[:] accum = self.__values  # View of the underlying location for the data
+        cdef double[:] values
+        cdef int i
+        cdef int nparam
+        cdef int n = accum.shape[0]
+        cdef ScenarioIndex scenario_index
+
         if self._agg_func == AggFuncs.PRODUCT:
-            value = 1.0
+            accum[...] = 1.0
             for parameter in self.parameters:
-                value *= parameter.get_value(scenario_index)
+                values = parameter.get_all_values()
+                for i in range(n):
+                    accum[i] *= values[i]
         elif self._agg_func == AggFuncs.SUM:
-            value = 0
+            accum[...] = 0.0
             for parameter in self.parameters:
-                value += parameter.get_value(scenario_index)
+                values = parameter.get_all_values()
+                for i in range(n):
+                    accum[i] += values[i]
         elif self._agg_func == AggFuncs.MAX:
-            value = float("-inf")
+            accum[...] = float("-inf")
```

Comment:
We should probably just define `inf` at the start of the module (see below also)